float32 target_cmd
float32 current_limit